### PR TITLE
Remove dependency on xml-js library in core-frontend.

### PR DESCRIPTION
**Pull request by mdastous-bentley**

### CODE REVIEW

**Review by aruniverse — APPROVED** [review]
feel like you can clean up lot of the code wrt getElementTextContent, but changes look ok to me as is

File: core/frontend/src/tile/map/WmtsCapabilities.ts [hunk 1]
```diff
@@ -28,6 +27,17 @@ async function getXml(url: string, credentials?: RequestBasicCredentials): Promi
...
 
+const getElementTextContent = (elem: Element, qualifiedName: string, defaultText?: string) => {
+  let text: string|undefined, found = false;
+  const tmpElem = elem.getElementsByTagName(qualifiedName);
+  if (tmpElem.length > 0) {
+    text = tmpElem[0].textContent ?? defaultText;
+    found = true;
+  }
+
+  return {found, text};
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply core/frontend/src/tile/map/WmtsCapabilities.ts:38]: First reason is that textContent type's is `string | null`, so this function defaults it to undefined.  Also if you provide defaultText, it becomes more difficult to know if a value has been found.

File: core/frontend/src/tile/map/WmtsCapabilities.ts [hunk 2]
```diff
@@ -97,29 +107,42 @@ export namespace WmtsCapability {
     public readonly title?: string;
     public readonly keywords?: string[];
 
-    constructor(json: any) {
-      this.abstract = json[OwsConstants.ABSTRACT_XMLTAG]?._text;
-      this.serviceType = json[OwsConstants.SERVICETYPE_XMLTAG]?._text;
-      this.serviceTypeVersion = json[OwsConstants.SERVICETYPEVERSION_XMLTAG]?._text;
-      this.title = json[OwsConstants.TITLE_XMLTAG]?._text;
+    constructor(elem: Element) {
+      const abstract = getElementTextContent(elem, OwsConstants.ABSTRACT_XMLTAG);
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply core/frontend/src/tile/map/WmtsCapabilities.ts:111]: string enum would look better I agree, I will make the change.

File: core/frontend/src/tile/map/WmtsCapabilities.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aruniverse) [reply core/frontend/src/tile/map/WmtsCapabilities.ts:38]: i was thinking you could just do something like
```ts
return textContent || defaultText
```
looks like most place where you provide a default text its an empty string, falsey 
i didnt look at every single of the 36 calls you have to getElementTextContent so might be missing the point, so feel free to close this comment as its more of a nit

**Review by pmconne — APPROVED** [review]
Approving on behalf of display.

**Pull request merged**